### PR TITLE
Update codebase to ZAP 2.12

### PR DESCRIPTION
**Pull request by thc202**
Change all add-ons and `testutils` to use 2.12 (SNAPSHOT).
Update code accordingly (e.g. add annotations, address deprecations).

### CODE REVIEW

File: addOns/sse/src/main/java/org/zaproxy/zap/extension/sse/EventStreamProxy.java [hunk 1]
```diff
@@ -71,14 +71,11 @@ public enum State {
     private ServerSentEventStream dataStreamObject;
 
     public EventStreamProxy(
-            HttpMessage message,
-            BufferedReader reader,
-            BufferedWriter writer,
-            ZapGetMethod method) {
+            HttpMessage message, BufferedReader reader, BufferedWriter writer, Socket socket) {
         //		this.message = message;
```
Comment (kingthorin) [inline addOns/sse/src/main/java/org/zaproxy/zap/extension/sse/EventStreamProxy.java:75]: I know it's unrelated but drop this?

**Review by psiinon — APPROVED** [review]
Great work!

File: addOns/sse/src/main/java/org/zaproxy/zap/extension/sse/EventStreamProxy.java (hunk 1 above)
Comment (thc202) [reply addOns/sse/src/main/java/org/zaproxy/zap/extension/sse/EventStreamProxy.java:75]: Will do, also, address some some TODOs.

Comment (kingthorin) [reply addOns/sse/src/main/java/org/zaproxy/zap/extension/sse/EventStreamProxy.java:75]: Thanks!

**Comment by thc202** [comment]
https://github.com/zaproxy/zap-extensions/compare/d60d70bb9572cbde9e66dc3a2966d615f1a713a2..7a1ce18cf56f955e2d7848f113bd8e8898faf2c5

File: addOns/sse/src/main/java/org/zaproxy/zap/extension/sse/EventStreamProxy.java (hunk 1 above)
Comment (thc202) [reply addOns/sse/src/main/java/org/zaproxy/zap/extension/sse/EventStreamProxy.java]: Done.

**Review by ricekot — APPROVED** [review]
Thank you!

**Pull request merged**